src/sqlfluff/api/simple.py

```diff
@@ -123,8 +124,19 @@ def fix(
     linter = Linter(config=cfg)
 
     result = linter.lint_string_wrapped(sql, fix=True)
-    fixed_string = result.paths[0].files[0].fix_string()[0]
-    return fixed_string
+    if fix_even_unparsable is None:
+        fix_even_unparsable = cfg.get("fix_even_unparsable")
+    should_fix = True
+    if not fix_even_unparsable:
+        # If fix_even_unparsable wasn't set, check for templating or parse
+        # errors and suppress fixing if there were any.
+        _, num_filtered_errors = result.count_tmp_prs_errors()
+        result.discard_fixes_for_lint_errors_in_files_with_tmp_or_prs_errors()
+        if num_filtered_errors > 0:
+            should_fix = False
+    if should_fix:
+        sql = result.paths[0].files[0].fix_string()[0]
```

Comment:
Sorry, I'm still having trouble with this (head now hurts!)

Why do we need to call `discard_fixes_for_lint_errors_in_files_with_tmp_or_prs_errors` here now?

If there are PRS or TMP errors (whether `noqa`ed or not) then we will set `should_fix = False` so won't fix, so no reason to discard.
If there are no PRS or TMP errors, then no reason to discard.

Reply:
We don't. This is an optimization exposed by splitting the one function into two functions. Will change.

Reply:
Done!